### PR TITLE
Run3-hcx330 Make use of ESGetToken in DPGAnalysis/HcalTools

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Make use of ESGetToken in DPGAnalysis/HcalTools

#### PR validation:

Use test from private area

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by jfernan2** [comment]
+1

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 1]
```diff
@@ -1940,14 +2421,15 @@ void CMTRawAnalyzer::endJob() {
   h_nls_per_run->Fill(float(lscounterrun));
   h_nls_per_run10->Fill(float(lscounterrun10));
 
+  /*
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:2424]: Why these chunks of code have been commented out?
If not needed, can they be removed?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 2]
```diff
@@ -3,14 +3,493 @@
...
+const float bphi = 72.;
+const int zneta = 22;
+const int znphi = 18;
+//const float zbphi = 18.;
+const int npfit = 220;
+const float anpfit = 220.;  // for SiPM:
+
+/////////////////////////////////////////////////////////////////////////
+//  SERVICE FUNCTIONS --------------------------------------------------------
+/*
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:434]: Is this commented out code needed?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 3]
```diff
@@ -1962,6 +2444,7 @@ void CMTRawAnalyzer::endJob() {
   std::cout << "===== Start writing user histograms =====" << std::endl;
   //////////////////////////////////////////////////////////////////////   scaling of some histoes:
   ///////////////////////////////////////////->Write();
+  /*
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:2447]: Is this commented out code needed?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 4]
```diff
@@ -7372,7 +7858,7 @@ void CMTRawAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iS
 void CMTRawAnalyzer::beginJob() {
   if (verbosity > 0)
     cout << "========================   beignJob START   +++++++++++++++++++++++++++" << endl;
-  hOutputFile = new TFile(fOutputFileName.c_str(), "RECREATE");
+  //  hOutputFile = new TFile(fOutputFileName.c_str(), "RECREATE");
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:7861]: Is this commented out code needed?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 5]
```diff
@@ -7394,532 +7880,543 @@ void CMTRawAnalyzer::beginJob() {
...
-    h_error6_HO = new TH1F("h_error6_HO", " ", 5, 0., 5.);
-    h_error7_HO = new TH1F("h_error7_HO", " ", 5, 0., 5.);
-    h_amplError_HO = new TH1F("h_amplError_HO", " ", 100, -2., 98.);
-    h_amplFine_HO = new TH1F("h_amplFine_HO", " ", 100, -2., 98.);
-    h_mapDepth4Error_HO = new TH2F("h_mapDepth4Error_HO", " ", neta, -41., 41., nphi, 0., bphi);
-    h_fiber0_HO = new TH1F("h_fiber0_HO", " ", 10, 0., 10.);
-    h_fiber1_HO = new TH1F("h_fiber1_HO", " ", 10, 0., 10.);
-    h_fiber2_HO = new TH1F("h_fiber2_HO", " ", 40, 0., 40.);
-    h_repetedcapid_HO = new TH1F("h_repetedcapid_HO", " ", 5, 0., 5.);
+    //  ha2 = fs_->make<TH2F>("ha2"," ", neta, -41., 41., nphi, 0., bphi);
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:7883]: Is this commented out code needed?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:2424]: If you want I can remove them. They are legacy code

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc (hunk 5 above)
Comment (bsunanda) [reply DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:7883]: All the commented out code are left to remind the legacy code. Should they be removed? Please suggest

Comment (perrotta) [reply DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:7883]: If the legacy code is not supposed to be resurrected any more, better to remove and clean up
Alternatively, add at least a couple of further lines of comment that explains what those lines are intended to.

**Comment by perrotta** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**